src/view_tab.cpp

```diff
@@ -144,9 +145,32 @@ void View::DrawTabNote(DeviceContext *dc, LayerElement *element, Layer *layer, S
             y -= m_doc->GetDrawingUnit(staff->m_drawingStaffSize)
                 - m_doc->GetDrawingStaffLineWidth(staff->m_drawingStaffSize);
         }
+        // Center for German tablature
+        else if (staff->IsTabLuteGerman()) {
+            y -= m_doc->GetGlyphHeight(SMUFL_EC17_luteGermanAUpper, glyphSize, drawingCueSize) / 2;
+        }
 
         dc->SetFont(m_doc->GetDrawingSmuflFont(glyphSize, false));
         this->DrawSmuflString(dc, x, y, fret, HORIZONTALALIGNMENT_center, glyphSize);
+
+        // Add overline if required
+        // TODO is this the correct way to add an overline?
+        if (overline && !fret.empty()) {
+            const int lineWidth = m_doc->GetDrawingStaffLineWidth(staff->m_drawingStaffSize);
+            const int width = m_doc->GetGlyphWidth(fret.at(0), glyphSize, drawingCueSize);
+            const int x1 = x - width / 2;
+            const int x2 = x + (fret.size() - 1) * width + width * 8 / 10; // trim right hand overhang on last character
+            const int y1 = y + m_doc->GetGlyphHeight(fret.at(0), glyphSize, drawingCueSize) + lineWidth;
+            const int y2 = y1;
+
+            dc->SetPen(m_currentColour, ToDeviceContextX(lineWidth), AxSOLID);
```

Comment:
This looks weird. Why do you use `ToDeviceContextX` here?

Reply:
Here I was copying the line drawing code from view_page.cpp DrawStaffLines - without having a full understanding of the drawing primitives.  Please advise, thanks.

Reply:
This should be just `lineWidth`. Maybe not base it on `staffLineWidth` but another value like `lyricLineThickness`? See what fits best and add just a factor for convenience.